pages/index.tsx

```diff
@@ -1,70 +1,73 @@
+import Dialog from "@reach/dialog"
 import type { NextPage } from "next"
 import Head from "next/head"
 import Image from "next/image"
+import { useState } from "react"
+import LoginForm from "../components/LoginForm"
 import styles from "../styles/Home.module.css"
+import Button from "../utils/Button"
 
 const Home: NextPage = () => {
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-      </Head>
-      <link rel="icon" href="/favicon.ico" />
+  const [openModal, setOpenModal] = useState("none")
 
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.jss!</a>
-        </h1>
+  const handleLogin = () => {
+    setOpenModal("login")
+  }
 
-        <p className={styles.description}>
-          Get started by editing{" "}
-          <code className={styles.code}>pages/index.tsx</code>
-        </p>
+  const handleRegister = () => {
+    setOpenModal("register")
+  }
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+  const handleClose = () => {
+    setOpenModal("none")
+  }
 
-          <a href="https://nextjs.org/learn" className={styles.card}>
-            <h2>Learn &rarr;</h2>
-            <p>Learn about Next.js in an interactive course with quizzes!</p>
-          </a>
+  const login = (formData) => {
+    console.log("login", formData)
+  }
 
-          <a
-            href="https://github.com/vercel/next.js/tree/canary/examples"
-            className={styles.card}
-          >
-            <h2>Examples &rarr;</h2>
-            <p>Discover and deploy boilerplate example Next.js projects.</p>
-          </a>
+  const register = (formData) => {
+    console.log("register", formData)
+  }
 
-          <a
-            href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
-            className={styles.card}
-          >
-            <h2>Deploy &rarr;</h2>
-            <p>
-              Instantly deploy your Next.js site to a public URL with Vercel.
-            </p>
-          </a>
+  return (
+    <div className={"index-container"}>
+      <Image src={"/books.png"} alt="books" width={80} height={80} />
+      <h1>Bookshelf</h1>
+      <div className="button-container">
+        <Button onClick={handleLogin} text="Login" />
+        <Button onClick={handleRegister} text="Register" />
+      </div>
+      <Dialog
+        aria-label="Login form"
+        className="dialog"
+        isOpen={openModal === "login"}
```

Comment:
Connected to my comment above, this should be 
```suggestion
        isOpen={isModalOpen}
```